x-pack/plugins/monitoring/server/lib/cluster/get_index_patterns.ts

```diff
@@ -41,6 +42,61 @@ export function getIndexPatterns(
       };
     }, {}),
   };
-
   return indexPatterns;
 }
+
+export function getLegacyIndexPattern({ productType }: { productType: string }) {
+  let indexPattern = '';
+  switch (productType) {
+    case 'elasticsearch':
+      indexPattern = INDEX_PATTERN_ELASTICSEARCH;
+      break;
+    default:
+      throw new Error('invalid product type to create index pattern');
+  }
+  return indexPattern;
+}
+
+// ***** Move out of clusters??
+export function getDsIndexPattern({
+  type = DS_INDEX_PATTERN_METRICS,
+  productType,
+  datasets,
+  namespace = '*',
+}: {
+  type?: string;
+  datasets?: string[];
+  productType: string;
+  namespace?: string;
+}): string {
+  // if there is one dataset, include in the index pattern else use *.
+  // we cannot specify more than one particular dataset in a query because the list
+  // of indices could be too long
+  let datasetsPattern = '';
+  if (datasets) {
+    datasetsPattern = datasets.length === 1 ? `${productType}.${datasets[0]}` : '*';
+  } else {
+    datasetsPattern = `${productType}.*`;
+  }
+  return `${type}-${datasetsPattern}-${namespace}`;
+}
+
```

Comment:
gives you both index patterns for a given `productType`.  Couldn't think of a better name but this would be `elasticsearch`, `kibana`, etc that are used in the index names.